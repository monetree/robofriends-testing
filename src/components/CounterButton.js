import React, { Component } from 'react';

class CounterButton extends Component {
  constructor() {
    super();
    this.state = {
      count: 0
    }
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true;
    }
    return false;
  }

  updateCount = () => {
    this.setState(state => {
      return { count: state.count + 1 }
    })
  }

  render() {
    return (
          <button id='counter' color={this.props.color} onClick={this.updateCount} className="f4 ba bw1 ph3 pv2 mb2 dib pa3 bw2 ma2 blue shadow-5 pointer">
            Count: {this.state.count}
          </button>
    );
  }
}


export default CounterButton;
