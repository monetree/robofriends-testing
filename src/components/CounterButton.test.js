import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';



it('renders without crashing', () => {
  expect(shallow(<CounterButton />)).toMatchSnapshot();
});

it('correctly increments the counter', () => {
  const mockColor = 'red'
  const wrapper = shallow(<CounterButton color={mockColor} />)
  wrapper.find('[id="counter"]').simulate('click');
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 2 });
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 3 });
  wrapper.find('[id="counter"]').simulate('click');
  expect(wrapper.state()).toEqual({ count: 4 });
  expect(wrapper.props().color).toEqual('red')
})
