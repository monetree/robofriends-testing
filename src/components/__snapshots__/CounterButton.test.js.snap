// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Count: ",
        0,
      ],
      "className": "f4 ba bw1 ph3 pv2 mb2 dib pa3 bw2 ma2 blue shadow-5 pointer",
      "color": undefined,
      "id": "counter",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "Count: ",
      0,
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Count: ",
          0,
        ],
        "className": "f4 ba bw1 ph3 pv2 mb2 dib pa3 bw2 ma2 blue shadow-5 pointer",
        "color": undefined,
        "id": "counter",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Count: ",
        0,
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
